RenderStatePackager/src/RenderStatePackager.cpp

```diff
@@ -130,48 +124,59 @@ bool RenderStatePackager::Execute(RefCntAutoPtr<IArchiver> pArchive)
         if (!Result.load())
             LOG_ERROR_AND_THROW("Failed to create state objects");
 
-        for (auto& pResouce : Shaders)
-            m_Shaders.emplace(HashMapStringKey{pResouce->GetDesc().Name, false}, pResouce);
+        for (auto& pResource : Shaders)
+            m_Shaders.emplace(HashMapStringKey{pResource->GetDesc().Name, false}, pResource);
 
-        for (auto& pResouce : RenderPasses)
-            m_RenderPasses.emplace(HashMapStringKey{pResouce->GetDesc().Name, false}, pResouce);
+        for (auto& pResource : RenderPasses)
+            m_RenderPasses.emplace(HashMapStringKey{pResource->GetDesc().Name, false}, pResource);
 
-        for (auto& pResouce : ResourceSignatures)
-            m_ResourceSignatures.emplace(HashMapStringKey{pResouce->GetDesc().Name, false}, pResouce);
+        for (auto& pResource : ResourceSignatures)
+            m_ResourceSignatures.emplace(HashMapStringKey{pResource->GetDesc().Name, false}, pResource);
 
-        DynamicLinearAllocator Allocator{DefaultRawMemoryAllocator::GetAllocator()};
-
-        auto FindShader = [&](const char* Name) -> IShader* {
+        auto FindShader = [&](const char* Name) -> IShader* //
+        {
             if (Name == nullptr)
                 return nullptr;
 
             auto Iter = m_Shaders.find(Name);
             if (Iter == m_Shaders.end())
+            {
+                Result.store(false);
```

Comment:
This seems to be unnecessary as you set `Result` to false in the catch(...) handler